lib/elixir/test/elixir/code_normalizer/literals_test.exs

```diff
@@ -0,0 +1,304 @@
+      assert_same ~S["\x0A"]
+      assert_same ~S["f\a\b\ro"]
+      assert_same ~S["double \" quote"]
+    end
+
+    test "keeps literal new lines" do
+      assert_same """
+      "fo
+      o"
+      """
+    end
+
+    test "with interpolation" do
+      assert_same ~S["one #{} three"]
+      assert_same ~S["one #{2} three"]
+    end
+
+    test "with escaped interpolation" do
+      assert_same ~S["one\#{two}three"]
+    end
+
+    test "with escapes and interpolation" do
+      assert_same ~S["one\n\"#{2}\"\nthree"]
+    end
+  end
+
+  describe "charlists" do
+    test "without escapes" do
+      assert_same ~S['']
+      assert_same ~S[' ']
+      assert_same ~S['foo']
+    end
+
+    test "with escapes" do
+      assert_same ~S['f\a\b\ro']
+      assert_same ~S['single \' quote']
+    end
+
+    test "keeps literal new lines" do
+      assert_same """
+      'fo
+      o'
+      """
+    end
+
+    test "with interpolation" do
+      assert_same ~S['one #{2} three']
+    end
+
+    test "with escape and interpolation" do
+      assert_same ~S['one\n\'#{2}\'\nthree']
+    end
+  end
+
+  describe "string heredocs" do
+    test "without escapes" do
+      assert_same ~S'''
+      """
+      hello
+      """
+      '''
+    end
+
+    test "with escapes" do
+      assert_same ~S'''
+      """
+      f\a\b\ro
+      """
+      '''
+
+      assert_same ~S'''
+      """
+      multiple "\"" quotes
+      """
+      '''
+    end
+
+    test "with interpolation" do
+      assert_same ~S'''
+      """
+      one
+      #{2}
+      three
+      """
+      '''
+
+      assert_same ~S'''
+      """
+      one
+      "
+      #{2}
+      "
+      three
+      """
+      '''
+    end
+
+    test "nested with empty lines" do
+      assert_same ~S'''
+      nested do
+        """
+
+        foo
+
+
+        bar
+
+        """
+      end
+      '''
+    end
+
+    test "nested with empty lines and interpolation" do
+      assert_same ~S'''
+      nested do
+        """
+
+        #{foo}
+
+
+        #{bar}
+
+        """
+      end
+      '''
+
+      assert_same ~S'''
+      nested do
+        """
+        #{foo}
+
+
+        #{bar}
+        """
+      end
+      '''
+    end
+
+    test "with escaped new lines" do
+      assert_same ~S'''
+      """
+      one\
+      #{"two"}\
+      three\
+      """
+      '''
+    end
+  end
+
+  describe "charlist heredocs" do
+    test "without escapes" do
+      assert_same ~S"""
+      '''
+      hello
+      '''
+      """
+    end
+
+    test "with escapes" do
+      assert_same ~S"""
+      '''
+      f\a\b\ro
+      '''
+      """
+
+      assert_same ~S"""
+      '''
+      multiple "\"" quotes
+      '''
+      """
+    end
+
+    test "with interpolation" do
+      assert_same ~S"""
+      '''
+      one
+      #{2}
+      three
+      '''
+      """
+
+      assert_same ~S"""
+      '''
+      one
+      "
+      #{2}
+      "
+      three
+      '''
+      """
+    end
+  end
```

Comment:
We need some tests for sigils too. You can copy the string tests and use ~S to build the strings. :)